### PR TITLE
genesis-builder: implemented for all runtimes

### PR DESCRIPTION
**Pull request by michalkucharczyk**
This PR implements [`GenesisBuilder` API](https://github.com/paritytech/polkadot-sdk/blob/a414ea7515c9cdc81f1d12410e646afc148250e8/substrate/primitives/genesis-builder/src/lib.rs#L38) for all the runtimes in polkadot repo.

Step towards: paritytech/polkadot-sdk#25

### CODE REVIEW

File: polkadot/runtime/polkadot/Cargo.toml [hunk 1]
```diff
@@ -311,9 +313,13 @@ try-runtime = [
...
 # to make it smaller like logging for example.
-on-chain-release-build = [ "sp-api/disable-logging" ]
+on-chain-release-build = [ "disable-genesis-builder", "sp-api/disable-logging" ]
 
 # Set timing constants (e.g. session period) to faster versions to speed up testing.
 fast-runtime = []
 
 runtime-metrics = [ "runtime-parachains/runtime-metrics", "sp-io/with-tracing" ]
+
+# When enabled, the GenesisBuilder API will not be supprted, GenesisConfig shall be
```
Comment (ordian) [inline polkadot/runtime/polkadot/Cargo.toml:323]:
```suggestion
# When enabled, the GenesisBuilder API will not be supported, GenesisConfig shall be
```

File: polkadot/runtime/kusama/Cargo.toml [hunk 2]
```diff
@@ -342,9 +344,13 @@ try-runtime = [
...
 # to make it smaller like logging for example.
-on-chain-release-build = [ "sp-api/disable-logging" ]
+on-chain-release-build = [ "disable-genesis-builder", "sp-api/disable-logging" ]
 
 # Set timing constants (e.g. session period) to faster versions to speed up testing.
 fast-runtime = []
 
 runtime-metrics = [ "runtime-parachains/runtime-metrics", "sp-io/with-tracing" ]
+
+# When enabled, the GenesisBuilder API will not be supprted, GenesisConfig shall be
```
Comment (ordian) [inline polkadot/runtime/kusama/Cargo.toml:354]:
```suggestion
# When enabled, the GenesisBuilder API will not be supported, GenesisConfig shall be
```

**Review by ordian — APPROVED** [review]

**Review by skunert — APPROVED** [review]

**Review by lexnv — APPROVED** [review]

**Review by davxy — APPROVED** [review]

File: polkadot/runtime/polkadot/Cargo.toml [hunk 3]
```diff
@@ -311,9 +313,13 @@ try-runtime = [
...
+on-chain-release-build = [ "disable-genesis-builder", "sp-api/disable-logging" ]
 
 # Set timing constants (e.g. session period) to faster versions to speed up testing.
 fast-runtime = []
 
 runtime-metrics = [ "runtime-parachains/runtime-metrics", "sp-io/with-tracing" ]
+
+# When enabled, the GenesisBuilder API will not be supported, GenesisConfig shall be
+# stripped from the final binary
+disable-genesis-builder = []
```
Comment (bkchr) [inline polkadot/runtime/polkadot/Cargo.toml:325]: Not sure we need them. Please remove them.

**Review by bkchr — APPROVED** [review]
Please remove everywhere the `disable-genesis-builder`. Otherwise looks good.

File: polkadot/runtime/polkadot/Cargo.toml (hunk 3 above)
Comment (michalkucharczyk) [reply polkadot/runtime/polkadot/Cargo.toml:325]: If the size is not a concern, then we can keep serde + genesis builder within binary. On the other hand it is not used on-chain. Less code, maybe better - I mean some audits, attack vectors, etc...

Anyway, I am also not sure, so I'll remove it.

Comment (michalkucharczyk) [reply polkadot/runtime/polkadot/Cargo.toml]: 7d2b3f1798

**Comment by Polkadot-Forum** [comment]
This pull request has been mentioned on **Polkadot Forum**. There might be relevant details there:

https://forum.polkadot.network/t/polkadot-release-analysis-v1-2-0/4451/1

File: polkadot/runtime/polkadot/Cargo.toml (hunk 3 above)
Comment (michalkucharczyk) [reply polkadot/runtime/polkadot/Cargo.toml]: From what I recall we just wanted to keep it simple, and reduce number of features. What simplifies life when it comes to managing configurations.

I don't recall any technical reason why this should be enabled for production. IMO you can safely disable this.

Comment (RomarQ) [reply polkadot/runtime/polkadot/Cargo.toml]: Thanks for the reply 👍

**Pull request merged**